src/app/services/EloCalculationService.ts

```diff
@@ -0,0 +1,106 @@
+}
+
+@Injectable({
+    providedIn: 'root',
+})
+export class EloCalculationService {
+
+    public static getNewElos(eloEntry: EloEntry): EloInfoPair {
+        const normalEloDifferences: EloDifferences = EloCalculationService.getNormalEloDifferences(eloEntry);
+        const zeroNewElo: number = EloCalculationService.getActualNewElo(eloEntry.eloInfoPair.playerZero.currentElo,
+                                                                         normalEloDifferences.pointChangeForZero);
+        const oneNewElo: number = EloCalculationService.getActualNewElo(eloEntry.eloInfoPair.playerOne.currentElo,
+                                                                        normalEloDifferences.pointChangeForOne);
+        return {
+            playerZero: {
+                currentElo: zeroNewElo,
+                numberOfGamePlayed: eloEntry.eloInfoPair.playerZero.numberOfGamePlayed + 1,
+            },
+            playerOne: {
+                currentElo: oneNewElo,
+                numberOfGamePlayed: eloEntry.eloInfoPair.playerOne.numberOfGamePlayed + 1,
+            },
+        };
+    }
+    public static getNormalEloDifferences(eloEntry: EloEntry): EloDifferences {
+        const wZero: number = EloCalculationService.getWFrom(eloEntry.winner, Player.ZERO);
+        const wOne: number = EloCalculationService.getWFrom(eloEntry.winner, Player.ONE);
+        const playerZeroInfo: EloInfo = eloEntry.eloInfoPair.playerZero;
+        const playerOneInfo: EloInfo = eloEntry.eloInfoPair.playerOne;
+        const eloZero: number = playerZeroInfo.currentElo;
+        const eloOne: number = eloEntry.eloInfoPair.playerOne.currentElo;
+        const pZero: number = this.getWinningProbability(eloZero, eloOne);
+        const pOne: number = this.getWinningProbability(eloOne, eloZero);
+        const playerZeroK: number = EloCalculationService.getKFrom(playerZeroInfo.numberOfGamePlayed);
+        const playerOneK: number = EloCalculationService.getKFrom(playerOneInfo.numberOfGamePlayed);
+        return {
+            pointChangeForZero: EloCalculationService.getNormalEloDifference(playerZeroK, wZero, pZero),
+            pointChangeForOne: EloCalculationService.getNormalEloDifference(playerOneK, wOne, pOne),
+        };
+    }
+    public static getNormalEloDifference(K: number, W: number, P: number): number {
+        return K * (W - P);
+    }
+    public static getActualNewElo(oldElo: number, eloDifference: number): number {
+        let minimumFinalElo: number = 1;
+        if (eloDifference < 0) {
+            if (oldElo < 100) {
+                eloDifference = 0; // Not removing elo to player bellow 100
+            } else {
+                minimumFinalElo = 100; // Not making a player fall back bellow 100
+            }
+        }
+        const newElo: number = oldElo + eloDifference;
+        return Math.max(minimumFinalElo, newElo);
+    }
+    public static getWFrom(winner: 'ZERO' | 'ONE' | 'DRAW', player: Player): number {
```

Comment:
what is W ?

Reply:
W: 0= joueur a perdu, 0.5 il a drawé, 1= il a gagné.
C'est le nom des variables sur la page wikipédia :D !

Reply:
tu sais ajouter un commentaire de documentation ? Aussi, est-ce nécéssaire que ceci soit public ?